File: backend/src/main/java/com/senior/project/backend/domain/Job.java
package com.senior.project.backend.domain;

import lombok.*;

import java.util.Date;
import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Temporal;
import jakarta.persistence.TemporalType;

@Entity
@Data
// @ToString
@NoArgsConstructor
@AllArgsConstructor
@Generated
public class Job {
    
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    private String name;
    private String description;
    private String location;
    private boolean isCoop;

    @Temporal(value = TemporalType.DATE)
    private Date startDate;

    @Temporal(value = TemporalType.DATE)
    private Date endDate;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name="student_details_id")
    @JsonIgnore
    @ToString.Exclude
    private StudentDetails studentDetails;
}

File: frontend/src/domain/Job.ts
export interface JobJSON{
    id: string;
    name: string;
    location: string;
    description: string;
    startDate: Date;
    endDate: Date
    coop: boolean;
    studentDetailsID: string;
}

export class Job{
    constructor(json: JobJSON){
        this.id = json.id;
        this.name = json.name;
        this.location = json.location;
        this.description = json.description;
        this.startDate = new Date(json.startDate);
        this.endDate = new Date(json.endDate);
        this.isCoop = json.coop;
        this.studentDetailsID = json.studentDetailsID;
    }

    id: string;
    name: string;
    location: string;
    description: string;
    startDate: Date;
    endDate: Date
    isCoop: boolean;
    studentDetailsID: string;

    static makeEmpty(){
        return new Job({
            id: '',
            name: '',
            location: '',
            description: '',
            startDate: new Date(),
            endDate: new Date (),
            coop: false,
            studentDetailsID: ''
        })
    }
}